### ISSUE TITLE
Gallery format / `ImageGallery` class is deprecated

### ISSUE THREAD

**Issue by ardnet** [SemanticMediaWiki/SemanticResultFormats#170]
Hi, I'm having this problem as per mentioned in error message below:

```
Deprecated: Use of ImageGallery::__construct was deprecated in MediaWiki 1.22. [Called from SRF\Gallery::getResultText in /vagrant/mediawiki/extensions/SemanticResultFormats/formats/gallery/Gallery.php at line 66] in /vagrant/mediawiki/includes/debug/MWDebug.php on line 309
```

I'm using Semantic Result Format 2.3, Semantic MediaWiki 2.4.1, and MediaWiki 1.28.0-alpha

Please advise how to solve this as I'm stuck with this currently.
Thanks a lot in advance.

**Comment by kghbln** [comment]
@ardnet  Thanks for reporting.
@JeroenDeDauw  Since you were the original author of this format if I am not mistaken you will probably like to have a peep at this and provide a fix. I believe this will be very much appreciated.

**Comment by ardnet** [comment]
thx for the respond @kghbln 
Looking forward how it goes with this, thx!

**Comment by mwjames** [comment]
> Deprecated: Use of ImageGallery::__construct was deprecated in MediaWiki 1.22. [Called from SRF\Gallery::getResultText in /vagrant/mediawiki/extensions/SemanticResultFormats/formats/gallery/Gallery.php at line 66] in /vagrant/mediawiki/includes/debug/MWDebug.php on line 309

I'm guessing this to be related to [0, 1]. If someone wants to make a PR, feel free.

[0] https://github.com/wikimedia/mediawiki/blob/af7ffaeeecd1908351863c0e928b8c5dcfb033fc/includes/gallery/ImageGalleryBase.php#L123-L131

[1] https://github.com/SemanticMediaWiki/SemanticResultFormats/blob/master/formats/gallery/Gallery.php#L66

**Comment by s7eph4n** [comment]
Fixed in e02b1b9
